rest/util.go

```diff
@@ -126,11 +126,13 @@ func compareEtag(etag, baseEtag string) bool {
 	if etag == "" {
 		return false
 	}
-	if etag == baseEtag {
-		return true
-	}
-	if l := len(etag); l == len(baseEtag)+2 && l > 3 && etag[0] == '"' && etag[l-1] == '"' && etag[1:l-1] == baseEtag {
-		return true
+	if strings.HasPrefix(etag, "W/") {
+		if etag[2:] == baseEtag {
+			return true
```

Comment:
You are ignoring the quotes. This will never match.

Reply:
I see what you mean now! @Dragomir-Ivanov, do you want to do a quick PR where you handle the quotes?